packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -310,6 +312,10 @@ export default class OperatorModeStateService extends Service {
   updateSubmode(submode: Submode) {
     this.state.submode = submode;
     this.schedulePersist();
+
+    if (submode === Submodes.Code) {
+      this.matrixService.setLLMModelForCodeMode();
```

Comment:
For two repetitions it seems okay but needing this here and in the assistant panel suggests to me we need a hook or something like that that lets us be able to specify setup during a state change.